src/main/java/org/axonframework/springboot/nativex/AxonRuntimeHints.java

```diff
@@ -0,0 +1,238 @@
+                       }
+                       if (m.isAnnotationPresent(QueryHandler.class)) {
+                           Class<?> returnType = m.getReturnType();
+                           if (!Collection.class.isAssignableFrom(returnType)) {
+                               apiClasses.add(returnType);
+                           }
+                       }
+                   });
+    }
+
+    private Set<Class<?>> getClasses(ClassLoader classLoader, String packageName) {
+        InputStream stream = classLoader.getResourceAsStream(
+                packageName.replaceAll("[" + PACKAGE_SEPARATOR + "]", FOLDER_SEPARATOR)
+        );
+        assert stream != null;
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+        List<String> lines = reader.lines().toList();
+
+        Stream<Class<?>> current = lines.stream()
+                                        .filter(this::isClassFile)
+                                        .map(line -> getClass(line, packageName));
+
+        Stream<Class<?>> nested = lines.stream()
+                                       .filter(this::isPackageFolder)
+                                       .map(String::trim)
+                                       .map(child -> setChildPackageName(packageName, child))
+                                       .map(pName -> getClasses(classLoader, pName))
+                                       .flatMap(Set::stream);
+
+        return Stream.concat(current, nested).collect(Collectors.toSet());
+    }
+
+    private Class<?> getClass(String className, String packageName) {
+        try {
+            return Class.forName(
+                    packageName + PACKAGE_SEPARATOR + className.substring(0, className.lastIndexOf(PACKAGE_SEPARATOR)));
+        } catch (ClassNotFoundException e) {
+            return null;
+        }
+    }
+
+    private boolean isClassFile(String path) {
+        return path.endsWith(".class");
+    }
+
+    private boolean isPackageFolder(String path) {
+        return !isClassFile(path);
+    }
+
+    private String setChildPackageName(String parent, String child) {
+        return parent + PACKAGE_SEPARATOR + child;
+    }
+
+    private boolean isAxonClass(Class<?> clazz) {
+        return clazz.isAnnotationPresent(Aggregate.class) ||
+                clazz.isAnnotationPresent(ProcessingGroup.class) ||
+                clazz.isAnnotationPresent(Saga.class) ||
+                Arrays.stream(clazz.getMethods()).anyMatch(this::isAxonMethod);
+    }
+
+    private boolean isAxonMethod(Method method) {
+        return method.isAnnotationPresent(EventHandler.class) ||
+                method.isAnnotationPresent(QueryHandler.class) ||
+                method.isAnnotationPresent(CommandHandler.class) ||
+                method.isAnnotationPresent(EventSourcingHandler.class) ||
+                method.isAnnotationPresent(SagaEventHandler.class) ||
+                method.isAnnotationPresent(DeadlineHandler.class);
```

Comment:
By checking the old `AxonNativeHints`, I also see it adds Axon-annotations that are intended for parameters.
Like `@MetaDataValue` and `@Timeout`, for example. 
Don't we need to cover those too?

Reply:
It seems like it already knows enough from enabling reflection in the methods. At least `@MetaDataValue` and `@Timestamp` work without doing anything in addition.